test/flows/maternity_paternity_calculator_flow/paternity_calculator_flow_test.rb

```diff
@@ -0,0 +1,791 @@
+require "test_helper"
+require "support/flow_test_helper"
+require "support/flows/maternity_paternity_calculator_flow_test_helper"
+
+class MaternityPaternityCalculatorFlow::PaternityCalculatorFlowTest < ActiveSupport::TestCase
+  include FlowTestHelper
+  include MaternityPaternityCalculatorFlowTestHelper
+
+  setup { testing_flow MaternityPaternityCalculatorFlow }
+
+  context "question: leave_or_pay_for_adoption?" do
+    setup do
+      testing_node :leave_or_pay_for_adoption?
+      add_responses what_type_of_leave?: "paternity"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of employee_date_matched_paternity_adoption? for a 'yes' response" do
+        assert_next_node :employee_date_matched_paternity_adoption?, for_response: "yes"
+      end
+
+      should "have a next node of baby_due_date_paternity? for a 'no' response" do
+        assert_next_node :baby_due_date_paternity?, for_response: "no"
+      end
+    end
+  end
+
+  context "question: baby_due_date_paternity?" do
+    setup do
+      testing_node :baby_due_date_paternity?
+      add_responses what_type_of_leave?: "paternity",
+                    leave_or_pay_for_adoption?: "no"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of baby_birth_date_paternity?" do
+        assert_next_node :baby_birth_date_paternity?, for_response: "2020-11-01"
+      end
+    end
+  end
+
+  context "question: employee_date_matched_paternity_adoption?" do
+    setup do
+      testing_node :employee_date_matched_paternity_adoption?
+      add_responses what_type_of_leave?: "paternity",
+                    leave_or_pay_for_adoption?: "yes"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of padoption_date_of_adoption_placement?" do
```

Comment:
Nitpickiest of nitpicks, I understood what was meant by padoption, given the context, but might be slightly ambiguous at a glance

Reply:
Yup good point, I don't see that as something that is in the scope for me to change as part of writing these tests as it is the node name in the Smart Answer (and if I was to start trying to improve this where to stop 😬 )

https://github.com/alphagov/smart-answers/blob/e96abd0d1784866e9ae4f76e5cbdc66e2e311435/app/flows/maternity_paternity_calculator_flow/paternity_calculator_flow.rb#L60